SystemManager/Inc/SM.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * SM.hpp
+ *
+ * System Manager Header
+ *
+ * Created on: Nov 7, 2022
+ * Author(s): Gordon Fountain
+ */
+
+#ifndef ZPSW3_SM_HPP
+#define ZPSW3_SM_HPP
+
+#include <stdint.h>
+#include "cmsis_os.h"
+// #include "LOS_Link.hpp"
+// #include "LOS_Actuators.hpp"
+#include "SM_StateManager.hpp"
+#include "AM.hpp"
+
+class SystemState;
+
+namespace SM {
+    // Enumerates the current drone status
+    enum Drone_Operation_Mode{BOOT, DISARMED, GROUND_OPS, FLIGHT, FATAL_FAILURE = -1};
+    int AM_PERIOD_MS = 5; // TODO, figure out what these should be
+    int PM_PERIOD_MS = 5; // 200Hz.
+    int TM_PERIOD_SLOW_MS = 20; // 50Hz. All of these numbers should be decided.
+    int TM_PERIOD_OPERATION_MS = 5;
+}
+
+class SystemManager {
+    public:
+        SystemManager();
+        void execute();
+        void setState(SystemState& newState);
+    private:
+        SystemState* currentState;
+        SM::Drone_Operation_Mode operation_mode;
+        SystemManager();
+
+        // TODO PM, AM, TM instances here
+        AttitudeManager& attitude_manager;
+        // PathManager& attitude_manager;
+        // TelemetryManager& attitude_manager;
```

Comment:
I built all this thread stuff before AM/PM/TM actually had code merged, so wasn't sure. Interfaces would be better if they handle their own init/thread

Reply:
Coming back to this. I think SM will call init functions for the manager objects first. Then to make the operate it will run threads that call the operation function of the managers.